src/pages/blamefile.js

```diff
@@ -1,20 +1,92 @@
-import React from 'react';
+import React, { useState } from 'react';
+import Loading from '../components/Loading';
 import { fetchBlameFileData } from '../utils/blamefileQueries/blamefileDataFetcher';
+import { validateRepoStats } from '../utils/validation/validate';
+import Tables from '../components/Tables';
 
-const BlameFile = () => {
-    const handleSubmit = (e) => {
+function History(props) {
+    return <li> 
+        {props.startingLine}
+        {props.endingLine}
+        {props.age}
+        {props.url}
+        <ul>
+            {props.oid}
+            {props.name}
+            {props.committedDate}
+            {props.totalCount}
+        </ul>
+    </li>
+}
+
+function BlameFile() {
+    const [error, setErrors] = useState({ ownername: { status: false, message: "" }, reponame: { status: false, message: "" }, ownername: { status: false, message: "" }, filepath: { status: false, messasge: "" } })
+    const [form, setForm] = useState({ ownername: "CSC4350-TR", reponame: "GitHubGo", filepath: "src/App.js" });
```

Comment:
```suggestion
    const [form, setForm] = useState({ ownername: "CSC4350-TR", reponame: "GitHubGo", filepath: "https://github.com/CSC4350-TR/GitHubGo/tree/main/src/App.js" });
```
We will be accepting the link in the file path.
The advantages of using the above methods are:
* Validation
* Branch identification
* Absolute paths are better than relative paths (my opinion)